torchqdynamics/mesolve.py

```diff
@@ -23,6 +23,21 @@ def mesolve(
     parameters: Optional[Tuple[nn.Parameter, ...]] = None,
     solver: Optional[SolverOption] = None,
 ):
+    # Args:
+    #     H: (b_H?, n, n)
+    #     rho0: (b_rho0?, n, n)
+    #
+    # Returns:
+    #     (y_save, exp_save) with
+    #     - y_save: (b_H?, b_rho0?, len(t_save), n, n)
+    #     - exp_save: (b_H?, b_rho0?, len(exp_ops), len(t_save))
+
+    # batch H and rho0 by default
+    H_batched = H[None, ...] if H.dim() == 2 else H
```

Comment:
Nit: the use of `unsqueeze` would be more consistent with `squeeze` below

Reply:
Right, I chose the current syntax because it's what we do in `odeint`, I think it's easier to understand what happens like that than with `unsqueeze`. Unfortunately, I don't think there's such a shorthand syntax for `squeeze`.

Reply:
Actually there is, we could do `y_save[0]` but it's far less expressive than `y_save.squeeze(0)` ...

Reply:
I agree that we should use `squeeze` and `[None, ...]`. They are the most readable forms.